Sources/TuistKit/Commands/EditCommand.swift

```diff
@@ -0,0 +1,68 @@
+import Basic
+import Foundation
+import Signals
+import SPMUtility
+import TuistGenerator
+import TuistSupport
+
+class EditCommand: NSObject, Command {
+    // MARK: - Static
+
+    static let command = "edit"
+    static let overview = "Generates a temporary project to edit the project in the current directory"
+
+    // MARK: - Attributes
+
+    private let projectEditor: ProjectEditing
+    private let opener: Opening
+    private let pathArgument: OptionArgument<String>
+
+    // MARK: - Init
+
+    required convenience init(parser: ArgumentParser) {
+        self.init(parser: parser, projectEditor: ProjectEditor(), opener: Opener())
+    }
+
+    init(parser: ArgumentParser, projectEditor: ProjectEditing, opener: Opening) {
+        let subparser = parser.add(subparser: EditCommand.command, overview: EditCommand.overview)
+        pathArgument = subparser.add(option: "--path",
+                                     shortName: "-p",
+                                     kind: String.self,
+                                     usage: "The path to the directory whose project will be edited.",
+                                     completion: .filename)
+        self.projectEditor = projectEditor
+        self.opener = opener
+    }
+
+    func run(with arguments: ArgumentParser.Result) throws {
+        let path = self.path(arguments: arguments)
+
+        let xcodeprojPath = try projectEditor.edit(at: path, in: EditCommand.temporaryDirectory.path)
+        Printer.shared.print("Opening Xcode to edit the project. Press CTRL + C once you are done editing")
+        try opener.open(path: xcodeprojPath)
```

Comment:
The open command has an option to make it blocking 

`open -w`

Additionally there's a `-n` option to open a new Instance, this should allow quitting just that one instance of Xcode rather than any previously opened Xcode

https://github.com/bloomberg/tuist/commit/73915279d01029c807b17063cdf5d8065341423d#diff-a4c7a31fca795885ec7f936d4cd5aef0R45

This may help the use case here

Reply:
We should really use `xed .` to use the selected version of Xcode. We also may be able to prepend with `xcrun` to use the correct tools version

Reply:
Using `xed` would be the preferred way, Sadly couldn't get the `-w` (wait) option to work correctly, the process remains blocked even after quitting Xcode :/ 

That said, maybe we don't need to launch a blocking instance of Xcode it was just an idea :)